File: Homework_II/ML2_programming/evaluate.py
import numpy as np
target=np.genfromtxt('targets.csv')
acc=[]
indices=[]
for i in range(1,11):
    fold=np.genfromtxt('fold%d.csv'%i,delimiter=',',dtype=np.int)
    indices.extend(list(fold[:,0]))
    accuracy=sum(target[fold[:,0]-1]==fold[:,1])/fold.shape[0]
    acc.append(accuracy)

if len(set(indices))==len(indices) and len(indices)==target.shape[0]:
    print(np.array(acc).mean())
else:
    print('not a valid cross-validation')

File: Homework_II_LR/ML2_programming/evaluate.m
targets=csvread('targets.csv');
acc=zeros(1,10);
indices=zeros(1,size(targets,1));
for i=1:10
    fold=csvread(['fold',num2str(i),'.csv']);
    indices(fold(:,1))=indices(fold(:,1))+1;
    acc(i)=sum(targets(fold(:,1),:)==fold(:,2))/size(targets,1);
end
if sum(indices==1)==size(targets,1)
    fprintf('%f\n',mean(acc));
else
    fprintf('not a valid cross-validation\n');
end